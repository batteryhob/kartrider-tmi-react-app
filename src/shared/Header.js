import React from 'react';

const Header = ({ menu1, menu2, menu3, menu4 }) => {
  return (
    <header>
        <div className="contents">
            <i className="fas fa-home"></i>
            <div className="menu-center desktop">
                <ul>
                    <li className="active">
                     { menu1 }
                    </li>
                    <li>
                     { menu2 }
                    </li>
                    <li>
                     { menu3 }
                    </li>
                    <li>
                     { menu4 }
                    </li>
                </ul>
                <div className="underbar">
                </div>
            </div>
            <div className="menu-right">
                <i className="fas fa-search"></i>
                <i className="fas fa-bars"></i>
            </div>
        </div>
    </header>
  );
}

export default Header;
