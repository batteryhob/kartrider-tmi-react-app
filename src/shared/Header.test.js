import React from 'react'
import {render, fireEvent, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import Header from './Header';

describe('<Header />', () => {
  it('프롭스 확인', () => {
    const utils = render(<Header menu1="검색" menu2="랭킹" menu3="카트" menu4="트랙" />);
    utils.getByText('검색');
    utils.getByText('랭킹');
    utils.getByText('카트');
    utils.getByText('트랙');
  });
});
