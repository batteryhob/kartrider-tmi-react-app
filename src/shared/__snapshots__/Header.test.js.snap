// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div>
  <header>
    <div
      class="contents"
    >
      <i
        class="fas fa-home"
      />
      <div
        class="menu-center desktop"
      >
        <ul>
          <li
            class="active"
          >
            검색
          </li>
          <li>
            랭킹
          </li>
          <li>
            카트
          </li>
          <li>
            트랙
          </li>
        </ul>
        <div
          class="underbar"
        />
      </div>
      <div
        class="menu-right"
      >
        <i
          class="fas fa-search"
        />
        <i
          class="fas fa-bars"
        />
      </div>
    </div>
  </header>
</div>
`;
